<?xml version="1.0"?>
<launch>

  <arg name="PlumeSimNumber" default="11"/>

  <arg name="simNumber" default="1"/>
  <arg name="saveResults" default="False"/>

  <arg name="xminMap" default="0"/>
  <arg name="xmaxMap" default="10"/>
  <arg name="yminMap" default="0"/>
  <arg name="ymaxMap" default="10"/>

  <param name="startTest" type="bool"  value="True" />

  <param name="simNumber" type="int"  value="$(arg simNumber)" />
  <param name="saveResults" type="bool"  value="$(arg saveResults)" />

  # Create simulated sensor
  <node pkg="tf" type="static_transform_publisher" name="ROBOTS2GADEN" args="0.0 0.0 0.0 0 0 0 map_gaden world 100" />
  <node pkg="tf" type="static_transform_publisher" name="Robot_to_sensor1" args="0.0 0.0 0.0 0 0 0 Robot_1/base_link pid0_frame 50" />

  <group ns="Robot_1">
    <include file="$(find gaden_sims)/launch/raster_PF_Gaden10x10.launch">
      <arg name="stayTime" default="0.01"/>
      <arg name="xminRaster"     default="9.75"/>
      <arg name="xmaxRaster"     default="0.25"/>
      <arg name="yminRaster"     default="0.25"/>
      <arg name="ymaxRaster"     default="9.75"/>
      <arg name="flipXY"         default="True"/>
    </include>

    <node pkg="simulated_gas_sensor" type="simulated_gas_sensor" name="fake_pid" output="log">
      <param name="/sensor_model" value="31" />
      <param name="/sensor_frame" value="pid0_frame" />
      <param name="/fixed_frame" value="map_gaden"/>
    </node>
  </group>

  # Plot particles
  <node name="plotMultiPlumePF" pkg="gaden_sims" type="plotMultiPlumePF.py" output="screen">
    <param name="xmin"     type="double"  value="$(arg xminMap)"    /> # m
    <param name="xmax"     type="double"  value="$(arg xmaxMap)"    /> # m
    <param name="ymin"     type="double"  value="$(arg yminMap)"    /> # m
    <param name="ymax"     type="double"  value="$(arg ymaxMap)"    /> # m
  </node>

  # Plot readings
  <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot1"  args="/Robot_1/Sensor_reading/raw"/>


  # Start plume sims!
  <!-- <include file="$(find plume_sims)/launch/gaden7.launch"/> -->

  # 1 plume
  <include file="$(find plume_sims)/launch/gaden11.launch" if="$(eval PlumeSimNumber == 11)"/>
  <include file="$(find plume_sims)/launch/gaden12.launch" if="$(eval PlumeSimNumber == 12)"/>
  <include file="$(find plume_sims)/launch/gaden13.launch" if="$(eval PlumeSimNumber == 13)"/>

  # 2 plumes
  <include file="$(find plume_sims)/launch/gaden21.launch" if="$(eval PlumeSimNumber == 21)"/>
  <include file="$(find plume_sims)/launch/gaden22.launch" if="$(eval PlumeSimNumber == 22)"/>
  <include file="$(find plume_sims)/launch/gaden23.launch" if="$(eval PlumeSimNumber == 23)"/>

  # 3 plumes
  <include file="$(find plume_sims)/launch/gaden31.launch" if="$(eval PlumeSimNumber == 31)"/>
  <include file="$(find plume_sims)/launch/gaden32.launch" if="$(eval PlumeSimNumber == 32)"/>
  <include file="$(find plume_sims)/launch/gaden33.launch" if="$(eval PlumeSimNumber == 33)"/>

  # 4 plumes
  <include file="$(find plume_sims)/launch/gaden41.launch" if="$(eval PlumeSimNumber == 41)"/>
  <include file="$(find plume_sims)/launch/gaden42.launch" if="$(eval PlumeSimNumber == 42)"/>
  <include file="$(find plume_sims)/launch/gaden43.launch" if="$(eval PlumeSimNumber == 43)"/>

</launch>
